bin/imsim.py

```diff
@@ -98,24 +100,24 @@ def main():
                               m5=[defaults.m5(bandpass)],
                               seeing=[seeing])
 
-    # Trim the input catalog to a single chip.
-    raICRS = phoSimObjectList['ra'].values
-    decICRS = phoSimObjectList['dec'].values
-    phoSimObjectList['chipName'] = chipNameFromRaDec(raICRS, decICRS,
-                                                     camera=camera,
-                                                     obs_metadata=obs,
-                                                     epoch=2000.0)
     # Now further sub-divide the source dataframe into stars and galaxies.
     if arguments.sensor is not None:
+        # Trim the input catalog to a single chip.
+        raICRS = phoSimObjectList['ra'].values
+        decICRS = phoSimObjectList['dec'].values
+        phoSimObjectList['chipName'] = chipNameFromRaDec(raICRS, decICRS,
+                                                         camera=camera,
+                                                         obs_metadata=obs,
+                                                         epoch=2000.0)
         starDataBase = \
-            phoSimObjectList.query("galSimType == 'pointSource' and magNorm < 50 and chipName=='%s'" % arguments.sensor)
+            phoSimObjectList.query("galSimType=='pointSource' and magNorm<50 and chipName=='%s'" % arguments.sensor)
```

Comment:
In its output summary, phosim lists the numbers of sources and the numbers of photons they generated by magnitude range.  For the Twinkles runs, no source with magnitude > 37 generates any recorded photons, and only ~10 sources that faint or fainter are in the instance catalogs (see, e.g., [this log file](http://srs.slac.stanford.edu/Pipeline-II/exp/LSST-DESC/log.jsp?pi=41888463) for ObsHistID 200.